server/src/routes/schedule.js

```diff
@@ -180,4 +178,129 @@ router.put('/interviewer/delete/:id', (req, res) => {
   });
 });
 
+// find all the possible meeting times, given the following constraints/information: 
+// attendess, timeConstraints, meetingDuration, locationConstraints
+router.post('/meeting', notAuthMiddleware, async (req, res) => {
+  // const id = 1;
+  // const meetingDuration = "PT1H";
+  const { candidate, meetingDuration, required, optional } = req.body;
+  // query the database to get the candidates availability (this will be useed as the time constratint)
+  const sql = 'SELECT * FROM Candidate c INNER JOIN CandidateAvailability a ON c.id = a.candidateID WHERE c.email = ? ORDER BY a.id DESC';
+  const sqlcmd = connection.format(sql, [candidate]);
+
+  connection.query(sqlcmd, async (err, result) => {
+    if (err) {
+      throw err;
+    }
+
+    console.log(result);
+
+    if (result.length === 0) {
+      res.send("No candidate availability found");
+    }
+    try {
+      const timeZone = "UTC";
+
+      const timeConstraint = {
+        activityDomain: "work",
+        timeSlots: result.map(time => ({
+          start: {
+            dateTime: time.startTime || new Date().toString(),
+            timeZone,
+          },
+          end: {
+            dateTime: time.endTime || new Date().toString(),
```

Comment:
not sure if we should initalize to new date() if we don't find any availability for our db for that candidate

I think we have a check (if result.length === 0)   -> "No candidate availability found"

Reply:
You're right! Just removed them